designs/multiple-worker-node-groups.md

```diff
@@ -0,0 +1,463 @@
+	  ...
+---
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineDeployment
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: eksa-test
+  name: eksa-test-md-0
+  namespace: eksa-system
+spec:
+  clusterName: eksa-test
+  replicas: 3
+  selector:
+    matchLabels: {}
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: eksa-test
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfigTemplate
+          name: eksa-test-md-0
+      clusterName: eksa-test
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: VSphereMachineTemplate
+        name: eksa-test-worker-node-template-1638469395669
+      version: v1.20.7-eks-1-20-8
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: VSphereMachineTemplate
+metadata:
+  name: eksa-test-worker-node-template-1638469395669
+  namespace: eksa-system
+spec:
+  template:
+    spec:
+    ...
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+kind: KubeadmConfigTemplate
+metadata:
+  name: eksa-test-md-1
+  namespace: eksa-system
+spec:
+  template:
+    spec:
+      joinConfiguration:
+        pause:
+          imageRepository: public.ecr.aws/eks-distro/kubernetes/pause
+          imageTag: v1.20.7-eks-1-20-8
+        bottlerocketBootstrap:
+          imageRepository: public.ecr.aws/l0g8r8j6/bottlerocket-bootstrap
+          imageTag: v1-20-8-eks-a-v0.0.0-dev-build.579
+        nodeRegistration:
+          criSocket: /var/run/containerd/containerd.sock
+          kubeletExtraArgs:
+            cloud-provider: external
+            tls-cipher-suites: Something
+          name: '{{ ds.meta_data.hostname }}'
+	  ...
+---
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineDeployment
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: eksa-test
+  name: eksa-test-md-1
+  namespace: eksa-system
+spec:
+  clusterName: eksa-test
+  replicas: 3
+  selector:
+    matchLabels: {}
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: eksa-test
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfigTemplate
+          name: eksa-test-md-1
+      clusterName: eksa-test
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: VSphereMachineTemplate
+        name: test
+      version: v1.20.7-eks-1-20-8
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: VSphereMachineTemplate
+metadata:
+  name: test
+  namespace: eksa-system
+spec:
+  template:
+    spec:
+    ...
+---
+```
+
+For each worker node groups, capi spec file will continue to have the following 3 kind fields.
+
+* KubeadmConfigTemplate
+* Machime deployment
+* VSphereMachineTemplate
+
+For each group, we will append these three fields corresponding to that group in the capi spec.
```

Comment:
What do you mean by `For each group, we will append these three fields`?
Like we will add 3 objects per worker node group?

Reply:
Yes